models/test/mapper/TestPreviewMapper.php

```diff
@@ -0,0 +1,202 @@
+<?php
+
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2020 (original work) Open Assessment Technologies SA ;
+ */
+
+declare(strict_types=1);
+
+namespace oat\taoQtiTestPreviewer\models\test\mapper;
+
+use oat\generis\model\OntologyAwareTrait;
+use oat\oatbox\service\ConfigurableService;
+use oat\taoQtiTestPreviewer\models\test\TestPreviewConfig;
+use oat\taoQtiTestPreviewer\models\test\TestPreviewMap;
+use qtism\data\AssessmentTest;
+use qtism\data\IAssessmentItem;
+use qtism\data\NavigationMode;
+use qtism\runtime\tests\Route;
+use qtism\runtime\tests\RouteItem;
+
+class TestPreviewMapper extends ConfigurableService implements TestPreviewMapperInterface
+{
+    use OntologyAwareTrait;
+
+    public function map(AssessmentTest $test, Route $route, TestPreviewConfig $config): TestPreviewMap
```

Comment:
quite big,  needs refactoring in the next step

Reply:
Yes, I avoided to touch this structure too much, cause was based on original test map and I did not want to take the risk at the end of the sprint. I agree can be reduced and we have test coverage, so now it will be easy.